### PR TITLE
 cleanup(gittag, githubrelease) remove deprecated code for field `version`

### PR DESCRIPTION
**Pull request by dduportal**
This PR removes the deprecated code of the field `version`.

It also introduces the following changes:

- Minor change on the `Makefile` to allow executing the end to end test with a custom binary directory (e.g. allow `make test-e2e VENOM_VAR_binpath=./custom-dist/` to use the binary `updatecli` located in `./custom-dist/updatecli`)
- Decouple the SCM handlers from the concept of source. It removes th "source" argument from the method `Init()` (interface ScmHandler) as it was only used to define versionfilters.

Associated documentation PR: https://github.com/updatecli/website/pull/264

## Test

To test this pull request, you can run the following commands:

```shell
make test
```

### CODE REVIEW

**Review by lemeurherve — DISMISSED** [review]

**Review by lemeurherve — APPROVED** [review]

**Comment by dduportal** [comment]
- I'm waiting for review from 2 maintainers on this one (to be really sure that I did not do something wrong).
- Proposed milestone: 0.22.3

File: pkg/plugins/resources/githubrelease/main.go
Comment (olblak) [inline pkg/plugins/resources/githubrelease/main.go:12]: Not blocker for this PR, but as recently discussed, It would be good to put comment before a struct field and not on the same line.

Comment (dduportal) [reply pkg/plugins/resources/githubrelease/main.go:12]: Yep, that would be a separated (and strictly scoped) PR to avoid messing up our rebases ;) But thanks to remind it to us 👍

**Review by olblak — APPROVED** [review]
Thanks for this huge cleanup, especially considering that I failed to correctly document the code which evolved many times

**Comment by dduportal** [comment]
> Thanks for this huge cleanup, especially considering that I failed to correctly document the code which evolved many times

Thanks, but don't be too harsh with yourself :) It's a looot of work, so let's improve this tool and its doc one step after the other

**Pull request merged**
